options.go

```diff
@@ -0,0 +1,56 @@
+package car
+
+import "math"
+
+// Options holds the configured options after applying a number of
+// Option funcs.
+//
+// This type should not be used directly by end users; it's only exposed as a
+// side effect of Option.
+type Options struct {
+	TraverseLinksOnlyOnce bool
+	MaxTraversalLinks     uint64
+}
+
+// Option describes an option which affects behavior when
+// interacting with the  interface.
+type Option func(*Options)
+
+// TraverseLinksOnlyOnce prevents the traversal engine from repeatedly visiting
+// the same links more than once.
+//
+// This can be an efficient strategy for an exhaustive selector where it's known
+// that repeat visits won't impact the completeness of execution. However it
+// should be used with caution with most other selectors as repeat visits of
+// links for different reasons during selector execution can be valid and
+// necessary to perform full traversal.
+func TraverseLinksOnlyOnce() Option {
+	return func(sco *Options) {
+		sco.TraverseLinksOnlyOnce = true
+	}
+}
+
+// MaxTraversalLinks changes the allowed number of links a selector traversal
+// can execute before failing.
+//
+// Note that setting this option may cause an error to be returned from selector
+// execution when building a SelectiveCar.
+func MaxTraversalLinks(MaxTraversalLinks uint64) Option {
+	return func(sco *Options) {
+		sco.MaxTraversalLinks = MaxTraversalLinks
+	}
+}
+
+// ApplyOptions applies given opts and returns the resulting Options.
+// This function should not be used directly by end users; it's only exposed as a
+// side effect of Option.
+func ApplyOptions(opt ...Option) Options {
+	opts := Options{
+		TraverseLinksOnlyOnce: false,         // default: recurse until exhausted
```

Comment:
I'd certainly defer to your judgment on the default value for this option. 
Thinking out loud I would have thought a typical user would want to set this to `true`, and maybe default `true` (or changing the name so that zero value translates to traversing only once) would be more reasonable?

Reply:
I see great points thank you